clock.go

```diff
@@ -2,58 +2,62 @@ package engo
 
 import (
 	"time"
-
-	"github.com/luxengine/math"
 )
 
+// The amound of nano seconds in a second.
+const secondsInNano int64 = 1000000000
+
 // A Clock is a measurement built in `engo` to measure the actual frames per seconds (framerate).
 type Clock struct {
-	elapsed float32
-	delta   float32
-	fps     float32
-	frames  uint64
-	start   time.Time
-	frame   time.Time
+	counter   uint32
+	perSecond uint32
+
+	deltaStamp int64
+	elapsStamp int64
+	frameStamp int64
+	startStamp int64
 }
 
 // NewClock creates a new timer which allows you to measure ticks per seconds. Be sure to call `Tick()` whenever you
 // want a tick to occur - it does not automatically tick each frame.
 func NewClock() *Clock {
+	currStamp := time.Now().UnixNano()
+
 	clock := new(Clock)
-	clock.start = time.Now()
-	clock.Tick()
+	clock.frameStamp = currStamp
+	clock.startStamp = currStamp
 	return clock
 }
 
 // Tick indicates a new tick/frame has occurred.
 func (c *Clock) Tick() {
-	now := time.Now()
-	c.frames += 1
-	if !c.frame.IsZero() {
-		c.delta = float32(now.Sub(c.frame).Seconds())
-	}
+	currStamp := time.Now().UnixNano()
+
+	c.counter += 1
 
-	c.elapsed += c.delta
-	c.frame = now
+	c.deltaStamp = currStamp - c.frameStamp
+	c.frameStamp = currStamp
 
-	if c.elapsed >= 1 {
-		c.fps = float32(c.frames)
-		c.elapsed = math.Mod(c.elapsed, 1)
-		c.frames = 0
+	c.elapsStamp += c.deltaStamp
+	if secondsInNano <= c.elapsStamp {
+		c.elapsStamp %= secondsInNano
+		c.perSecond = c.counter
+		c.counter = 0
 	}
 }
 
 // Delta is the amount of seconds between the last tick and the one before that
 func (c *Clock) Delta() float32 {
-	return c.delta
+	return float32(float64(c.deltaStamp) / float64(secondsInNano))
```

Comment:
Even though a `float64` divided by a `float64` is faster than using `float32` (because of some weird optimizaiton in the Go library), this might not be the case when you have to cast it to `float32` again? 

Not sure though, but just a thought.